Assets/HoloToolkit/Input/Scripts/Interactable.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Use this class for easy access to OnTap, OnGazeEnder, and OnGazeExit events.
+    /// Be sure to override each of these methods with your own implimentation.
+    /// Examples of code usage in each method below in comments.
+    /// </summary>
+    [RequireComponent(typeof(Collider))]
+    public class Interactable : MonoBehaviour
+    {
+        protected virtual void OnEnable()
+        {
+            GestureManager.Instance.OnTap += OnTap;
+            GazeManager.Instance.OnGazeExit += OnGazeExit;
+            GazeManager.Instance.OnGazeEnter += OnGazeEnter;
+        }
+
+        protected virtual void OnDisable()
+        {
+            GestureManager.Instance.OnTap -= OnTap;
+            GazeManager.Instance.OnGazeExit -= OnGazeExit;
+            GazeManager.Instance.OnGazeEnter -= OnGazeEnter;
+        }
+
+        /// <summary>
+        /// Called when a user has tapped any GameObject.
+        /// </summary>
+        /// <param name="tappedGameObject">GameObject user has tapped.</param>
+        protected virtual void OnTap(GameObject tappedGameObject)
+        {
+            // Do something if ANY GameObject has been tapped.
```

Comment:
I really don't think all those commented lines are needed here. The documentation + your example scripts should be enough.